### PR TITLE
Add Jreleaser configuration and clean up build script

### PR DESCRIPTION
**Pull request by hpmellema**
*Description of changes:*
Implements a number of changes to improve the release process for the plugins. 
1. Removes unnecessary source and javadoc jar tasks (the `com.gradle.plugin-publish` plugin already generates theses so they were redundant)
2. Removes the `maven-publish` setup as the `com.gradle.plugin-publish` plugin already creates publications.
3. Renames subprojects to match generated artifacts (otherwise plugin-publish will generate additional unused artifacts)
4. Updates the build to use Jreleaser for actually publishing artifacts to align with the release process used by the primary smithy repo.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rschmitt** [comment]
Does `--configuration-cache` work now?

**Comment by hpmellema** [comment]
Configuration caching should be supported as of 0.8.0. 
See: https://github.com/smithy-lang/smithy-gradle-plugin/issues/56
A test case was also added here: https://github.com/smithy-lang/smithy-gradle-plugin/blob/65e0271eac1477f0e911ef1a6bdfc7d24300da92/smithy-jar-plugin/src/it/java/software/amazon/smithy/gradle/SourceProjectionTest.java#L41

**Comment by rschmitt** [comment]
I was asking about the build of `smithy-gradle-plugin` itself, more out of curiosity than anything. It looks like configuration caching doesn't work, even with your changes:

```
* What went wrong:
Configuration cache state could not be cached: field `isSupportingMultipleReports` of task `:spotbugsMain` of type `com.github.spotbugs.snom.SpotBugsTask`: error writing value of type 'org.gradle.api.internal.provider.DefaultProvider'
> Could not resolve all files for configuration ':spotbugs'.
   > Cannot resolve external dependency com.github.spotbugs:spotbugs:4.7.3 because no repositories are defined.
     Required by:
         project :
```

Note, however, that this is not due to a configuration caching _problem_; it's because there's a dependency closure that for some reason isn't getting resolved:

```
[1] ~/src/smithy-gradle-plugin # g dependencies --configuration spotbugs

> Configure project :
JReleaser Gradle Plugin 1.9.0. Consider becoming a sponsor at https://opencollective.com/jreleaser
Smithy Gradle version: '0.9.0'
Kotlin DSL property assignment is an incubating feature.

> Task :dependencies

------------------------------------------------------------
Root project 'smithy-gradle'
------------------------------------------------------------

spotbugs - configuration for the SpotBugs engine
\--- com.github.spotbugs:spotbugs:4.7.3 FAILED
```

Maybe you could fix this by not applying the spotbugs plugin to the root project?

**Comment by rschmitt** [comment]
I think the easiest way to fix CC is to define repositories in the root project:

```diff
     id("com.adarshr.test-logger") version "3.2.0"
     id("com.gradle.plugin-publish") version "1.2.1" apply false
     id("org.jreleaser") version "1.9.0"
 }

+repositories {
+    mavenLocal()
+    mavenCentral()
+}
+
```

The plan is to enable CC by default some time this year, so you might as well get it working now.

File: build.gradle.kts [hunk 1]
```diff
@@ -256,4 +216,55 @@ subprojects {
...
 
+    // Configuration for deploying to Maven Central.
+    // https://jreleaser.org/guide/latest/examples/maven/maven-central.html#_gradle
+    deploy {
+        maven {
+            nexus2 {
+                create("maven-central") {
+                    active.set(Active.ALWAYS)
+                    url.set("https://s01.oss.sonatype.org/service/local")
+                    snapshotUrl.set("https://s01.oss.sonatype.org/content/repositories/snapshots/")
```
Comment (JordonPhillips) [inline build.gradle.kts:257]: Pretty sure we should be using the AWS urls here

**Review by JordonPhillips — CHANGES_REQUESTED** [review]

**Review by JordonPhillips — APPROVED** [review]

**Pull request merged**